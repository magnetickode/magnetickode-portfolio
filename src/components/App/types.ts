import { MutableRefObject } from "react";

export interface Props {
  navbarRef: MutableRefObject<HTMLDivElement | null>;
}
