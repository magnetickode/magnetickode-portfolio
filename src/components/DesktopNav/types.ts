export interface Props {
  sticky: boolean;
}

export interface StyledDesktopNavProps {
  sticky: boolean;
}

export interface StyledNavLinkProps {
  sticky: boolean;
}
