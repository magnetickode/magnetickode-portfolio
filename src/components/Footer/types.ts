export interface Props {
  footerState: string;
  footerPosition: string;
}

export interface StyledFooterProps {
  footerState: string;
  footerPosition: string;
}
