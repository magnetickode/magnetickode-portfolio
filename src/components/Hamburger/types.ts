export interface Props {
  opened: boolean;
  toggleHamburger: () => void;
}

export interface StyledHamburgerProps {
  opened: boolean;
}
