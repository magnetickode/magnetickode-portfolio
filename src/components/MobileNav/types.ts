export interface Props {
  collapse: boolean;
  hashRoute: string;
}

export interface StyledMobileNavProps {
  collapse: boolean;
}

export interface StyledNavLinkProps {
  active: boolean;
}
