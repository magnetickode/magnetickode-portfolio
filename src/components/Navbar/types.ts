export interface Props {
  hamburgerOpened: boolean;
  mobileNavVisible: boolean;
  toggleHamburger: () => void;
  isDesktop: boolean;
  stickyNavVisible: boolean;
  hashRoute: string;
  changeTheme: () => void | null;
}

export interface StyledNavbarProps {
  sticky: boolean;
}
