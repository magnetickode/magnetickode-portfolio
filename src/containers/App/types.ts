import { ScreenSizeChangedGen, ShowStickyNavGen } from "../../store/actions/types";

export interface Props {
  isDesktop: boolean;
  themeIndex: number;
  stickyNavVisible: boolean;
  screenSizeChanged: ScreenSizeChangedGen;
  showStickyNav: ShowStickyNavGen;
}
