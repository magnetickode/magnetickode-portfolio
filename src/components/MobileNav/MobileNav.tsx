import React from "react";
import styled, { keyframes } from "styled-components";

import { Props, NavItemProps } from "./types";

const enlargeAnim = keyframes`
  from {
    height: 0;
  } 

  to {
    height: 22.5rem;
  }
`;

const StyledMobileNav = styled.div`
  position: absolute;
  top: 6.9rem;
  right: 2.8rem;
  width: 16rem;
  background: ${({ theme }) => theme.textColor};
  border-radius: 0.5rem;
  box-shadow: 0 0 2rem rgba(0, 0, 0, 0.5);
  user-select: none;
  overflow: hidden;
  animation: ${enlargeAnim} 0.5s;
`;

const NavItem = styled.div<NavItemProps>`
  width: 100%;
  height: 4.5rem;
  box-sizing: border-box;
  display: flex;
  align-items: center;
  padding-left: 2rem;
  font-weight: bold;
  background: ${({ sticky, theme }) => (sticky ? theme.primaryColor : theme.textColor)};
  color: ${({ sticky, theme }) => (sticky ? theme.textColor : theme.primaryColor)};
  transition: all 0.35s;
  cursor: pointer;

  :hover {
    background: ${({ sticky, theme }) => (sticky ? theme.textColor : theme.primaryColor)};
    color: ${({ sticky, theme }) => (sticky ? theme.primaryColor : theme.textColor)};
  }
`;

const MobileNav: React.FC<Props> = ({ sticky }) => (
  <StyledMobileNav data-test="MobileNavComponent">
    <NavItem sticky={sticky}>Home</NavItem>
    <NavItem sticky={sticky}>Skills</NavItem>
    <NavItem sticky={sticky}>Portfolio</NavItem>
    <NavItem sticky={sticky}>About</NavItem>
    <NavItem sticky={sticky}>Contact</NavItem>
  </StyledMobileNav>
);

export default MobileNav;
