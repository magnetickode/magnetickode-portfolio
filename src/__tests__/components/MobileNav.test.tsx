import React from "react";
import { shallow, ShallowWrapper } from "enzyme";

import MobileNav from "../../components/MobileNav";
import { Props } from "../../components/MobileNav/types";

let wrapper: ShallowWrapper;

const props: Props = {
  sticky: false,
  collapse: false
};

beforeEach(() => {
  wrapper = shallow(<MobileNav {...props} />);
});

it("matches snapshot", () => {
  expect(wrapper).toMatchSnapshot();
});

it("renders without error", () => {
  const MobileNavComponent = wrapper.find("[data-test='MobileNavComponent']");
  expect(MobileNavComponent.length).toBe(1);
});
