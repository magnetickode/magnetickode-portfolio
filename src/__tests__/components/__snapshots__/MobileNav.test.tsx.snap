// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<styled.div
  collapse={false}
  data-test="MobileNavComponent"
>
  <styled.div
    active={false}
  >
    <NavHashLink
      exact={true}
      smooth={true}
      to="/#home"
    >
      Home
    </NavHashLink>
  </styled.div>
  <styled.div
    active={false}
  >
    <NavHashLink
      smooth={true}
      to="/#skills"
    >
      Skills
    </NavHashLink>
  </styled.div>
  <styled.div
    active={false}
  >
    <NavHashLink
      smooth={true}
      to="/#portfolio"
    >
      Portfolio
    </NavHashLink>
  </styled.div>
  <styled.div
    active={false}
  >
    <NavHashLink
      smooth={true}
      to="/#about"
    >
      About
    </NavHashLink>
  </styled.div>
  <styled.div
    active={false}
  >
    <NavHashLink
      smooth={true}
      to="/#contact"
    >
      Contact
    </NavHashLink>
  </styled.div>
</styled.div>
`;
